### PR TITLE
feat(schemas): logs schema and log payload type

### PR DESCRIPTION
**Pull request by IceHe**
## Summary

- Add `logs` schema
- Add `LogPayload` type

References:

- Varchar index length: https://stackallflow.com/database/is-it-a-good-idea-approach-to-index-a-varchar-column/

> There’s nothing wrong with indexing a varchar column if you’re going to be doing queries based on it. However please keep in mind that there a limits to some indexes and how much they can index in a single field. Example you can’t index a column that can contain an unlimited amount of text. However, you **should be able to do an index on `varchar(256)` without issue**. Try it, and analyze the improvements in your queries performance to see if it helps.

```sql
create index type__route__status_code on logs (type, route, status_code);
```
224 < type + status + status = 226 < 256

## Linear Issue Reference

LOG-899
LOG-2147
LOG-2193

## Testing

No need for now

### CODE REVIEW

File: packages/schemas/src/foundations/jsonb-types.ts [hunk 1]
```diff
@@ -72,6 +73,23 @@ export const userLogPayloadGuard = z.object({
...
+
+export const logPayloadGuard = z.object({
+  ip: z.string().optional(),
+  userAgent: z.string().optional(),
+  applicationId: z.string().optional(),
+  connectorId: z.string().optional(),
+  userId: z.string().optional(),
+  request: z.object({ body: z.object({}).optional() }).optional(),
+  response: z.object({ body: z.object({}).optional() }).optional(),
+  // TODO: Add necessary fields when needed (LOG-2193)
```
Comment (gao-sun) [inline packages/schemas/src/foundations/jsonb-types.ts:88]: is these fields and comment necessary now? we can remove until needed

Comment (gao-sun) [reply packages/schemas/src/foundations/jsonb-types.ts:88]: don't add more to-do comments unless we have to

**Review by gao-sun — CHANGES_REQUESTED** [review]

File: packages/schemas/tables/logs.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  type        varchar(32)                 not null,
+  route       varchar(192)                not null,
+  status_code int2                        not null,
+  payload     jsonb /* @use LogPayload */ not null default '{}'::jsonb,
+  created_at  timestamptz                 not null default (now()),
+  primary key (id)
+);
+
+create index type__route__status_code on logs (type, route, status_code);
+create index created_at on logs (created_at);
```
Comment (gao-sun) [inline packages/schemas/tables/logs.sql:13]: these indices do not match current convention

File: packages/schemas/src/foundations/jsonb-types.ts [hunk 3]
```diff
@@ -60,6 +60,7 @@ export type Identities = z.infer<typeof identitiesGuard>;
 
 /**
  * User Logs
+ * TODO: DEPRECATED, and remove later
  */
 
 export const userLogPayloadGuard = z.object({
```
Comment (demonzoo) [inline packages/schemas/src/foundations/jsonb-types.ts:66]: use `@deprecated` annotation

File: packages/schemas/src/foundations/jsonb-types.ts (hunk 1 above)
Comment (IceHe) [reply packages/schemas/src/foundations/jsonb-types.ts]: Removed. PTAL

File: packages/schemas/src/foundations/jsonb-types.ts (hunk 3 above)
Comment (IceHe) [reply packages/schemas/src/foundations/jsonb-types.ts:66]: Updated. PTAL

Comment (gao-sun) [reply packages/schemas/src/foundations/jsonb-types.ts:66]: deprecated already means "remove later"

File: packages/schemas/src/foundations/jsonb-types.ts [hunk 4]
```diff
@@ -72,6 +73,22 @@ export const userLogPayloadGuard = z.object({
...
+ */
+
+export const logPayloadGuard = z.object({
+  ip: z.string().optional(),
+  userAgent: z.string().optional(),
+  applicationId: z.string().optional(),
+  connectorId: z.string().optional(),
+  userId: z.string().optional(),
+  request: z.object({ body: z.object({}).optional() }).optional(),
+  response: z.object({ body: z.object({}).optional() }).optional(),
```
Comment (gao-sun) [inline packages/schemas/src/foundations/jsonb-types.ts:87]: i also meant `request` and `response` fields

File: packages/schemas/src/foundations/jsonb-types.ts (hunk 3 above)
Comment (IceHe) [reply packages/schemas/src/foundations/jsonb-types.ts:66]: Removed. PTAL

File: packages/schemas/src/foundations/jsonb-types.ts (hunk 4 above)
Comment (IceHe) [reply packages/schemas/src/foundations/jsonb-types.ts:88]: See Slack thread [discussion](https://silverhand-io.slack.com/archives/C02CMB8JVJB/p1650016973075049)

**Comment by IceHe** [comment]
Conclusion:
- just record user logs and ignore management logs for v1.0
    - remove `route` and `status code` fields
- will extract an appropriate abstract after implementing the concrete user logs
    - guard `payload` field as an arbitrary object

See Slack thread [discussion](https://silverhand-io.slack.com/archives/C02CMB8JVJB/p1650016973075049)

**Review by gao-sun — APPROVED** [review]

**Pull request merged**